sdks/python/apache_beam/runners/worker/worker_pool_main.py

```diff
@@ -0,0 +1,177 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Worker pool entry point.
+
+The worker pool exposes an RPC service that is used with EXTERNAL
+environment to start and stop the SDK workers.
+
+The worker pool uses child processes for parallelism; threads are
```

Comment:
It is a bit odd to read this here because `BeamFnExternalWorkerPoolServicer` supports threads internally and tests make use of this feature. If this module is the entry point for the worker pool, maybe the pooling code itself should be located in a different module and we should only handle its configuration here.

Reply:
The documentation applies to the entry point, which always uses processes. It's OK if tests use the inner class separately, it should have its own documentation (I did not modify that part, just moved it).